crates/polars-core/src/utils/supertype.rs

```diff
@@ -395,6 +395,7 @@ pub fn get_supertype_with_options(
                     None => None
                 }
             },
+            (dt, String) if dt.is_categorical() => Some(dt.clone()),
```

Comment:
`Unknown(UnknownKind::Str)` was already supported, see the next section of the `match` statement.